### PR TITLE
Adding the POST /api/sources/:source_id/check_availability action

### PR DESCRIPTION
**Pull request by abellotti**
Adding the POST `/api/sources/:source_id/check_availability` action
- Action triggers an availability check on the Source[:source_id] specified
- Returns a 202 with empty {} body for now.

### CODE REVIEW

**Comment by abellotti** [comment]
marking as wip, needs some more testing and specs.

File: app/controllers/api/v1/sources_controller.rb [hunk 1]
```diff
@@ -15,6 +15,23 @@ def create
...
+        source = Source.find(params[:source_id])
+
+        Sources::Api::Messaging.client.publish_topic(
+          :service => "platform.topological-inventory.operations-#{source.source_type.name}",
+          :event   => "Source.availability_check",
+          :payload => {
+            :params => {
+              :source_id       => source.id.to_s,
+              :external_tenant => source.tenant.external_tenant
+            }
```
Comment (agrare) [inline app/controllers/api/v1/sources_controller.rb]: We should create a task so the requester can track the progress of the availability check, pass the task ID in as a param, and return the ID to the caller

Comment (agrare) [reply app/controllers/api/v1/sources_controller.rb]: Scratch that, we're going to hold off on the UI blocking on the availability check

**Comment by abellotti** [comment]
moved back to wip, not sure we want to return the source, if future enhancements may be to include tasks,  thinking of just the 202/no_content for this.  @agrare ?

**Comment by agrare** [comment]
@abellotti yeah that sounds good to me

File: lib/sources/api/messaging.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
+module Sources
+  module Api
+    module Messaging
+      def self.client
+        require "manageiq-messaging"
+
+        @client ||= ManageIQ::Messaging::Client.open(
```
Comment (agrare) [inline lib/sources/api/messaging.rb]: Can you update `Sources::Api::Events` to use this messaging client?

**Pull request merged**